pkg/target/http_test.go

```diff
@@ -342,33 +345,44 @@ func TestHttpWrite_Simple(t *testing.T) {
 
 			var results [][]byte
 			wg := sync.WaitGroup{}
-			server := createTestServerWithResponseCode(&results, &wg, tt.ResponseCode)
+			server := createTestServerWithResponseCode(&results, tt.ResponseCode)
 			defer server.Close()
 
-			target, err := newHTTPTarget(server.URL, 5, 1, 1048576, 1048576, "application/json", "", "", "", "", "", "", true, false, "", "", "", "")
+			target, err := newHTTPTarget(server.URL, 5, tt.BatchSize, 1048576, 1048576, "application/json", "", "", "", "", "", "", true, false, "", "", "", "")
 			if err != nil {
 				t.Fatal(err)
 			}
 
 			var ackOps int64
 			ackFunc := func() {
 				atomic.AddInt64(&ackOps, 1)
+
+				wg.Done()
 			}
 
-			messages := testutil.GetTestMessages(501, "Hello Server!!", ackFunc)
-			wg.Add(501)
+			messages := testutil.GetTestMessages(200, "Hello Server!!", ackFunc)
+			wg.Add(200)
 			writeResult, err1 := target.Write(messages)
 
-			wg.Wait()
+			if ok := WaitForAcksWithTimeout(2*time.Second, &wg); !ok {
+				assert.Fail("Timed out waiting for acks")
+			}
 
 			assert.Nil(err1)
-			assert.Equal(501, len(writeResult.Sent))
-			assert.Equal(501, len(results))
+			assert.Equal(200, len(writeResult.Sent))
+			assert.Equal(200/tt.BatchSize, len(results))
```

Comment:
So here we check number of batches sent, but I think we never don't check later what is the actual size of each sent batch, correct? 

I think it would be nice to have this additional assertion.

Also maybe add another case for not equal batches, like: batch size - 3, msg count - 7, out - 3 x 2 + 1?
Aaand maybe...it would be easier to not write this test as one of the case of this `...Simple` test but add another separate one, focusing on batching?

Doesn't really matter, this one is also fine :)

Reply:
Good points. Pushed a change